src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1646,7 +1646,10 @@ public override void VisitMarkupTagHelperStartTag(MarkupTagHelperStartTagSyntax
             {
                 foreach (var child in node.Attributes)
                 {
-                    if (child is MarkupTagHelperAttributeSyntax || child is MarkupMinimizedTagHelperAttributeSyntax)
+                    if (child is MarkupTagHelperAttributeSyntax ||
+                        child is MarkupMinimizedTagHelperAttributeSyntax ||
+                        child is MarkupTagHelperDirectiveAttributeSyntax ||
+                        child is MarkupMinimizedTagHelperDirectiveAttributeSyntax)
```

Comment:
So I'm guessing that the conclusion of this is that parser knows about directive attributes now? 😁 

It feels so good to do actual language design 👍

Reply:
Well technically, only the tag helper rewriter phase know about it but yes the syntax tree now has first class support for directive attributes.